.github/workflows/build_deb.yaml

```diff
@@ -0,0 +1,234 @@
+        required: false
+        type: string
+        default: ''
+
+env:
+  AUTHOR: "Guillaume Raffin"
+  MAIL: "theelectronwill@gmail.com"
+  ALUMET_DIR: ${{ inputs.name }}-${{ inputs.version }}
+
+jobs:
+  packaging:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["debian:11", "debian:12", "ubuntu:20.04", "ubuntu:22.04", "ubuntu:24.04", "ubuntu:25.04"]
+
+    container:
+      image: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-tags: 'true'
+
+      - name: Set dynamic environment variables
+        run: |
+          echo "OS_NAME=$(grep '^ID=' "/etc/os-release" | cut -d'=' -f2 | tr -d '"')" >> $GITHUB_ENV
+          echo "OS_VERSION=$(grep '^VERSION_ID=' "/etc/os-release" | cut -d'=' -f2 | tr -d '"')" >> $GITHUB_ENV
+
+      # Installation of dependencies required for the package compilation
+      - name: Install dependencies
+        shell: bash
+        run: |
+          apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y build-essential devscripts debhelper
+
+      # Rename the alumet-X.X path with correct version before to enter in it
+      - name: Alumet source code downloading from GitHub repository
+        run: |
+          base=$(find . -maxdepth 1 -type d -name "${{ inputs.name }}-*")
+
+          if [ -d "$base" ]; then
+              if [ ! -d "${{ env.ALUMET_DIR }}" ]; then
+                  mv -f "$base" "${{ env.ALUMET_DIR }}"
+              fi
+          else
+              echo "\e[31mERROR\e[0m : No directory '$dir' found."
+              exit 1
+          fi
+
+      # Get correct source code version on Github with tag
+      - uses: actions/checkout@v4
+        with:
+          repository: alumet-dev/alumet
+          fetch-tags: 'true'
+          ref: '${{ inputs.tag }}'
+          path: '${{ env.ALUMET_DIR }}/alumet'
+
+      - name: Install Rust and dependencies
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y || exit
+
+          echo '. "$HOME/.cargo/env"' >> $HOME/.bashrc
+          . "$HOME/.cargo/env"
+
+          packages=("curl" "git" "build-essential" "protobuf-compiler" "libssl-dev" "pkg-config")
+          for package in "${packages[@]}"; do
+            if ! dpkg -l | grep -q "^ii  $package "; then
+              echo -e "\e[33mWARNING\e[0m : Packages '$package' required. Installation..."
+              apt-get install -y "$package" || exit
+            fi
+          done
+
+          rustup default stable
+
+      - name: Compile Alumet
+        run: |
+          . "$HOME/.cargo/env"
+          base=$(pwd)
+          cd ${{ env.ALUMET_DIR }}/alumet/agent
+          mkdir -p ${base}/bin/
+          CARGO_TARGET_DIR=${base}/bin/ ALUMET_AGENT_RELEASE=true cargo build --release -p ${{ inputs.name }}-agent --bins --all-features
```

Comment:
maybe add argument to select the target architecture the binary will be compiled for. Or in case we want to hardcode x86_64/amd64 let's remove `arch` variable for now or make it hardly set to explicit that we do not manage multiple architectures for now ?